flytekit/common/nodes.py

```diff
@@ -216,13 +166,65 @@ def executable_sdk_object(self):
         return self._executable_sdk_object
 
     @classmethod
-    def promote_from_model(cls, model):
+    def promote_from_model(cls, model, sub_workflows=None, tasks=None):
         """
         :param flytekit.models.core.workflow.Node model:
+        :param list[flytekit.models.core.workflow.WorkflowTemplate] sub_workflows:
+        :param list[flytekit.models.task.TaskTemplate] tasks: If specified, these task templates will be passed to the
+            SdkTaskNode promote_from_model call, and used instead of fetching from Admin.
         :rtype: SdkNode
         """
-        raise _user_exceptions.FlyteAssertion("An SDK node cannot be instantiated merely from a data model object "
-                                              "because it must be contextualized within a workflow.")
+        id = model.id
+        # This should never be called
+        if id == _constants.START_NODE_ID or id == _constants.END_NODE_ID:
+            _logging.warning("Should not call promote from model on a start node or end node {}".format(model))
+            return None
+
+        sdk_task_node, sdk_workflow_node = None, None
+        if model.task_node is not None:
+            sdk_task_node = _component_nodes.SdkTaskNode.promote_from_model(model.task_node, tasks)
+        elif model.workflow_node is not None:
+            sdk_workflow_node = _component_nodes.SdkWorkflowNode.promote_from_model(
+                model.workflow_node,sub_workflows, tasks)
+        else:
+            raise _system_exceptions.FlyteSystemException("Bad Node model, neither task nor workflow detected")
+
+        # When WorkflowTemplate models (containing node models) are returned by Admin, they've been compiled with a
+        # start node.  In order to make the promoted SdkWorkflow look the same, we strip the start-node text back out.
+        for i in model.inputs:
+            if i.binding.promise is not None and i.binding.promise.node_id == _constants.START_NODE_ID:
```

Comment:
we should really tag nodes as system nodes or something when they are inserted by the compiler so we can avoid these types of checks. but for another day, i suppose